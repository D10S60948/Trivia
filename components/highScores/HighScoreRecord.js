import React, { Component } from 'react';
import { StyleSheet, View, Text } from 'react-native';

export default class HighScoreRecord extends Component {

    render() {
        return (
            <View style={styles.container} >
                <View style={{ ...styles.field, flex: 1, alignItems: 'center' }}>
                    <Text style={styles.text}>{this.props.ranking}</Text>
                </View>
                <View style={{ ...styles.field, flex: 8 }}>
                    <Text style={{...styles.text, paddingLeft: 10}}>{this.props.name}</Text> 
                </View>
                <View style={{ ...styles.field, flex: 2, alignItems: 'center' }}>
                    <Text style={styles.text}>{this.props.score}</Text>     
                </View>
            </View>
        );
    }
}

const styles = StyleSheet.create({
  container: {
    height: '8%',
    margin: '1%', 
    justifyContent: 'space-around',
    flexDirection: 'row'
  },
  field: {
    backgroundColor: 'rgba(0,51,102,0.25)',
    justifyContent: 'center',
    marginHorizontal: 10,
    borderRadius: 7
  },
  text: { 
    fontFamily: 'Delius', 
    fontSize: 16,
  }
});
