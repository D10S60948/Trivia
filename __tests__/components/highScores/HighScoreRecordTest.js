import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import HighScoreRecord from '../../../components/highScores/HighScoreRecord';

test('"HighScoreRecord" component renders correctly', () => {
    const index = 4;
    const value = {name: 'Test', score: 100};
    const snapshot = renderer.create(
        <HighScoreRecord ranking={ index + 1 } name={ value.name } score={ value.score } key={index} />
        ).toJSON();
    expect(snapshot).toMatchSnapshot();
});
