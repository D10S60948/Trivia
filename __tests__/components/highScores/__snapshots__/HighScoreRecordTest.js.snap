// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`"HighScoreRecord" component renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "height": "8%",
      "justifyContent": "space-around",
      "margin": "1%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(0,51,102,0.25)",
        "borderRadius": 7,
        "flex": 1,
        "justifyContent": "center",
        "marginHorizontal": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "fontFamily": "Delius",
          "fontSize": 16,
        }
      }
    >
      5
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "rgba(0,51,102,0.25)",
        "borderRadius": 7,
        "flex": 8,
        "justifyContent": "center",
        "marginHorizontal": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "fontFamily": "Delius",
          "fontSize": 16,
          "paddingLeft": 10,
        }
      }
    >
      Test
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(0,51,102,0.25)",
        "borderRadius": 7,
        "flex": 2,
        "justifyContent": "center",
        "marginHorizontal": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "fontFamily": "Delius",
          "fontSize": 16,
        }
      }
    >
      100
    </Text>
  </View>
</View>
`;
